File: DSAlgorithms/Leetcode/c++/arrays_hash/1-two_sum.cpp
std::vector<int> twoSum(std::vector<int>& nums, int target)
{
    std::unordered_map<int, int> hash;
    int vector_size = nums.size();
    for(int index = 0; index < vector_size; index++)
    {
        int complement = target - nums[index];
        if(hash.count(complement))
            return {hash[complement], index};
        hash[nums[index]] = index;
    }
    return {};
}

File: DSAlgorithms/Leetcode/ts/src/arrays_hash/1-two_sum.ts
const twoSum = (nums: number[], target: number): number[] =>{
    const hash = new Map<number, number>();
    for(let i = 0; i < nums.length; i++)
    {
        const complement = target - nums[i];
        if(hash.has(complement))
            return [hash.get(complement), i];
        hash.set(nums[i], i);
    }
    return [];
};
